frontend/pluginloader.lua

```diff
@@ -85,14 +93,64 @@ function PluginLoader:loadPlugins()
     end
 
     -- set package path for all loaded plugins
-    for _,plugin in ipairs(self.plugins) do
+    for _,plugin in ipairs(self.enabled_plugins) do
         package.path = string.format("%s;%s/?.lua", package.path, plugin.path)
         package.cpath = string.format("%s;%s/lib/?.so", package.cpath, plugin.path)
     end
 
-    table.sort(self.plugins, function(v1,v2) return v1.path < v2.path end)
+    table.sort(self.enabled_plugins, function(v1,v2) return v1.path < v2.path end)
+
+    return self.enabled_plugins, self.disabled_plugins
+end
+
+function PluginLoader:genPluginManagerSubItem()
+    local enabled_plugins, disabled_plugins = {}, {}
+    if self.all_plugins == nil then
+        self.all_plugins = {}
+        enabled_plugins, disabled_plugins = self:loadPlugins()
+    end
+
+    for _, plugin in ipairs(enabled_plugins) do
+        local element = {}
+        element.fullname = plugin.fullname or plugin.name
+        element.name = plugin.name
+        element.enable = true
+        table.insert(self.all_plugins, element)
+    end
+
+    for _, plugin in ipairs(disabled_plugins) do
+        local element = {}
+        element.fullname = plugin.fullname or plugin.name
+        element.name = plugin.name
+        element.enable = false
+        if element.name ~= "storagestat" then
+            table.insert(self.all_plugins, element)
+        end
+    end
+    table.sort(self.all_plugins, function(v1, v2) return v1.fullname < v2.fullname end)
 
-    return self.plugins
+    local plugin_table = {}
+    for __, plugin in ipairs(self.all_plugins) do
+        table.insert(plugin_table, {
+            text = plugin.fullname,
+            checked_func = function()
+                return plugin.enable
+            end,
+            callback = function()
+                local plugins_disabled = G_reader_settings:readSetting("plugins_disabled") or {}
+                plugins_disabled[plugin.name] = plugin.enable
+                plugin.enable = not plugin.enable
```

Comment:
That's an unclear shortcut :) to store the former value before the `not` , because we want `true` in the settings when it's becoming `false`.
But we don't want `false` in the settings when it's becoming `true`, we'd prefer `nil`, so it's not stored in the setting and we only have the list of disabled plugins in it (with all `=true`)
So, you may do that after the `not`, with a clearer `if plugin.enable then store nil (=erase) else store true end`